src/libtiled/hex.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * hex.h
+ * Copyright 2017, Benjamin Trotter <bdtrotte@ucsc.edu>
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "hex.h"
+
+using namespace Tiled;
+
+Hex::Hex(int x, int y, int z):
+    mX(x),
+    mY(y),
+    mZ(z) { }
+
+Hex::Hex(QPoint point,
+         Map::StaggerIndex staggerIndex,
+         Map::StaggerAxis staggerAxis)
+{
+    Hex(point.x(),point.y(),staggerIndex,staggerAxis);
```

Comment:
Thanks for this. I was thinking logically what I was doing wasn't quite right, but thought I would get an error if it was wrong. But I see now exactly what it was doing. 

> instead you could also define a Hex::setStaggered(int col, int row, ...)

Good advice! Implementing this way.